integration_tests/store_test.go

```diff
@@ -130,44 +126,74 @@ func (c *checkRequestClient) SendRequest(ctx context.Context, addr string, req *
 	return resp, err
 }
 
-func (s *testStoreSuite) TestRequestPriority(c *C) {
+func (s *testStoreSuite) TestRequestPriority() {
 	client := &checkRequestClient{
 		Client: s.store.GetTiKVClient(),
 	}
 	s.store.SetTiKVClient(client)
 
 	// Cover 2PC commit.
 	txn, err := s.store.Begin()
-	c.Assert(err, IsNil)
+	s.Require().Nil(err)
 	client.priority = kvrpcpb.CommandPri_High
 	txn.SetPriority(tikv.PriorityHigh)
 	err = txn.Set([]byte("key"), []byte("value"))
-	c.Assert(err, IsNil)
+	s.Nil(err)
 	err = txn.Commit(context.Background())
-	c.Assert(err, IsNil)
+	s.Nil(err)
 
 	// Cover the basic Get request.
 	txn, err = s.store.Begin()
-	c.Assert(err, IsNil)
+	s.Require().Nil(err)
 	client.priority = kvrpcpb.CommandPri_Low
 	txn.SetPriority(tikv.PriorityLow)
 	_, err = txn.Get(context.TODO(), []byte("key"))
-	c.Assert(err, IsNil)
+	s.Nil(err)
 
 	// A counter example.
 	client.priority = kvrpcpb.CommandPri_Low
 	txn.SetPriority(tikv.PriorityNormal)
 	_, err = txn.Get(context.TODO(), []byte("key"))
 	// err is translated to "try again later" by backoffer, so doesn't check error value here.
-	c.Assert(err, NotNil)
+	s.NotNil(err)
 
 	// Cover Seek request.
 	client.priority = kvrpcpb.CommandPri_High
 	txn.SetPriority(tikv.PriorityHigh)
 	iter, err := txn.Iter([]byte("key"), nil)
-	c.Assert(err, IsNil)
+	s.Nil(err)
 	for iter.Valid() {
-		c.Assert(iter.Next(), IsNil)
+		s.Nil(iter.Next())
 	}
 	iter.Close()
 }
+
+func (s *testStoreSuite) TestFailBusyServerKV() {
+	txn, err := s.store.Begin()
+	s.Require().Nil(err)
+	err = txn.Set([]byte("key"), []byte("value"))
+	s.Nil(err)
+	err = txn.Commit(context.Background())
+	s.Nil(err)
+
+	var wg sync.WaitGroup
+	wg.Add(2)
+
+	s.Nil(failpoint.Enable("tikvclient/rpcServerBusy", `return(true)`))
+	go func() {
+		defer wg.Done()
+		time.Sleep(time.Millisecond * 100)
```

Comment:
Shall we wait until the next goroutine done instead of depending on a "long enough" sleep?